### ISSUE TITLE
Why am I getting poor prediction results?

### ISSUE THREAD

**Issue by harryqt** [RubixML/Sentiment#5]
```bash
harry@ubuntu-server:~/sentiment$ php74 predict.php
Enter some text to analyze:
rubixml is great
The sentiment is: negative

harry@ubuntu-server:~/sentiment$ php74 predict.php
Enter some text to analyze:
Rubix ML is really great
The sentiment is: positive
```

**Comment by andrewdalpino** [comment]
Hahahhaa hilarious @Dibbyo456 ... ohhhhhh machine learning you! But yeah there can be many reasons for poor real-world performance even on seemingly obvious (for us) examples. What is the validation score your model achieves after training? Were you able to visualize training loss and validation score at each epoch?

The first thing I'd do is try to identify if your model is underfit or overfit. Also note that the model is trained on movie reviews (usually about a couple paragraphs) and therefore is biased (see selection bias) somewhat toward larger blobs of text.

Did you modify the training script at all or did you stick with the default settings?

**Comment by harryqt** [comment]
> Hahahhaa hilarious.... ohhhhhh machine learning you!

🤣

> Did you modify the training script at all or did you stick with the default settings?

The only change I did was `AdaMax` to `Adam` because of Tensor extension issue as seen on #4 

About all the other questions you asked, I have no idea.. But I will try to figure them out.. hopefully.

**Comment by harryqt** [comment]
![msedge_2021-05-21_10-02-24](https://user-images.githubusercontent.com/18257605/119082485-d7a91d80-ba1b-11eb-84a0-1effb1e36ef1.png)

![msedge_2021-05-21_10-02-33](https://user-images.githubusercontent.com/18257605/119082492-d8da4a80-ba1b-11eb-85cf-376d35283291.png)

**Comment by andrewdalpino** [comment]
Sounds good @Dibbyo456, I reminded the Zephir team of the issue just now. Unfortunately, due to a lack of resources we haven't had a chance to improve the demo model on rubixml.com in a very long time - it's likely that someone with any time budget and access to compute will be able to create a more accurate model.

Let us know how well your model performs during CV and training I'll be happy to help you further from  there.
